### ISSUE TITLE
[rfc] func-style: allow arrow functions?

### ISSUE THREAD

**Issue by kaelig** [Shopify/eslint-plugin-shopify#158]
I was wondering what folks thought of relaxing the func-style rule (currently set to: `['error', 'declaration']`), like so:

```json5
      "func-style": [
        "error",
        "declaration",
        {
          "allowArrowFunctions": true
        }
      ]
```

It still frowns upon this code:

```js
// still invalid
var foo = function() {
    // ...
};
```

But would allow arrow functions, which can be useful in a functional-style program.

```js
// would be valid with allowArrowFunctions: true
var foo = () => {};
```

See https://eslint.org/docs/rules/func-style

---

Please :+1: the issue if you’re for it, or :-1: with a comment explaining why if against it :heart:

**Comment by GoodForOneFare** [comment]
I think the reason for holding out on this was named functions being easier to identify in stack traces.  Seems like less of an issue now that we have first class sourcemaps everywhere.

**Comment by kaelig** [comment]
Closed by https://github.com/Shopify/eslint-plugin-shopify/pull/188
